### DISCUSSION TITLE
Azure DevOps output handler

### DISCUSSION THREAD

**Discussion by knackstedt** [avajs/ava#3277] [Q&A]
Please provide details about:

#### What I'm trying to do:
I want to use AVA to run tests in Azure DevOps. The problem, is that Azure DevOps has no real ANSI color support, so the color codes get garbled up and the output becomes a mess.
![image](https://github.com/avajs/ava/assets/29349167/35eb4e6b-c63f-4517-97ee-6fa40b56f798)

Azure DevOps has a strange syntax for pipeline output, and it would be nice if there would be an output adapter for it so my output is actually readable. 
https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash#formatting-commands

I'm willing to work on this, given the time and help from people who know what I need to do to get started and what the quality criteria are. I'm not a huge open source contributor so please bear with me :sweat:

**Comment by novemberborn** [comment]
You can set the environment variable `FORCE_COLOR=0`, or run `ava --no-color`, or set `color: false` in the AVA configuration.

**Reply by knackstedt** [reply]
That can disable the color, but it won't make use of Azure DevOps additional functionality for grouping console output and marking errors with their syntax
This was more of my question / idea -- what would it take to set a flag that uses a different output format that uses Azure DevOps special syntax?

**Reply by novemberborn** [reply]
That's possible but highly experimental. See the code in https://github.com/avajs/ava/pull/3247 which shipped in AVA 6. You can get the internal state of a test run and then write that to disk, so that your next step in CI can ship it to Azure DevOps.
